[GUI] Create QR-code for Masternode private key

crowning- opened the pull request:
@QuantumExplorer asked for an enhancement to export the private key of a Masternode to an external device via QR-code.
I've added this to the "My Masternodes" tab where you can select a Masternode and open a screen with the QR-code of its private key.

Looks like this:
![show_key](https://user-images.githubusercontent.com/10080039/36948842-fac07008-1fe0-11e8-96c0-92b65d2788f1.png)

Some additional information:
- I think we will have more and more use cases where it's convenient  to export data via QR-code, that's why I created a dedicated class + UI for it
- the existing QR-export for requesting payments has data processing logic and presentation layer all mixed up, that's very inflexible. My approach has the data processing where it belongs (and where the data actually is, here: masternodelist.cpp), and the QR-UI just gets an appropriate screen title, the data to be converted to QR-code, and a (HTML-) string with the additional text as parameters.
- compared to the payment-request screen I've removed the copy-to-clipboard buttons, they would reduce flexibility (you would have to know which part of the text is what) and the usual mark + copy&paste via mouse does the same job.
- the dialog for payment-requests should be refactored/removed to use the same logic, but that's out of scope for this PR and can be done in a dedicated PR somewhere in the future.
- the screenshot shows an unused private key, so you can't steal my valuable tDASH ;-)

Reviews/feedback welcome.

crowning- commented:
I'll have a look today after work.

Another option would be to leave the QR-code popup as it is, but provide the additional information via double-click on one of the Masternodes in that list. But that would need another class + .ui.

I think I'll add your changes and we can discuss probable usability issues from there .

UdjinM6 commented:
FYI: pushed another small fix to my branch https://github.com/UdjinM6/dash/commit/4b4ccac750830cef5a72ae9b154dde43a5928888 (otherwise versions are going to be confusing until we upgrade to 0.12.3)

crowning- commented:
Integrated your changes and squashed.

For me it wasn't obvious which of all these informations is displayed as QR-code, so I added a dedicated label above the QR-code to make that clear:

![qr-new](https://user-images.githubusercontent.com/10080039/36994820-88f8c1be-20b2-11e8-9cf0-27f82c5650c6.png)

UdjinM6 commented:
mnprivkey _should_ hold no funds, so your tDASH _should_ be safe :) but one can (re)start your MN using this info (I won't, promise :D)

jhodges10 commented:
Any reason it's just the Private Key that's exposed in the QR code?

Would be nice to be able to add a node to an external application for management with just the QR code.
